### PR TITLE
build(aio): fail doc-gen if a code-example is badly formatted.

### PR DESCRIPTION
**Pull request by petebacondarwin**
This will catch the problem that was missed in
https://github.com/angular/angular/pull/19845#issuecomment-338626662

### CODE REVIEW

**Comment by petebacondarwin** [comment]
Yay!
https://travis-ci.org/angular/angular/jobs/291545055#L1142

```

```

**Comment by petebacondarwin** [comment]
See here: https://pr19870-2ba4cb4.ngbuilds.io/guide/ajs-quick-reference#ng-href

screen shot 2017-10-23 at 15 13 46

**Comment by gkalpak** [comment]
Not directly related to this PR, but these suggestions are still relevant (just saying 😛).

**Comment by petebacondarwin** [comment]
You mean the suggestions on the issue?

File: aio/tools/transforms/examples-package/processors/render-examples.spec.js [hunk 1]
```diff
@@ -34,10 +34,10 @@ describe('renderExamples processor', () => {
     describe(CODE_TAG, () => {
       it(`should ignore a <${CODE_TAG}> tags with no path attribute`, () => {
         const docs = [
-          { renderedContent: `Some text\n<${CODE_TAG}>Some code</${CODE_TAG}>\n<${CODE_TAG} class="anti-pattern" title="Bad Code">do not do this</${CODE_TAG}>` }
+          { renderedContent: `Some text\n<${CODE_TAG}>Some code</${CODE_TAG}>\n<${CODE_TAG} class="anti-pattern" title="Bad Code">do <strong>not</strong> do this</${CODE_TAG}>` }
```
Comment (petebacondarwin) [inline aio/tools/transforms/examples-package/processors/render-examples.spec.js]: put some HTML into the code example here to help show that this won't get affected by these new checks

File: aio/tools/transforms/examples-package/processors/render-examples.js [hunk 2]
```diff
@@ -12,16 +12,23 @@ module.exports = function renderExamples(getExampleRegion) {
...
-          });
+          doc.renderedContent = doc.renderedContent.replace(
+            /<(code-example|code-pane)([^>]*)>[^<]*<\/([^>]+)>/g,
+            (original, openingTag, attributes, closingTag) => {
+              const attrMap = parseAttributes(attributes);
+              if (attrMap.path) {
+                if (closingTag !== openingTag) {
+                  throw new Error(
+                    'Badly formed example: ' + original + ' - closing tag does not match opening tag.\n' +
+                    ' - Perhaps you forgot to put a blank line before the example?');
```
Comment (gkalpak) [inline aio/tools/transforms/examples-package/processors/render-examples.js]: Maybe add a short comment here explaining how this is related :smiley:
If one doesn't know what is going on, it really sounds irrelevant :smiley:
&nbsp;&ensp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;:smiley:

**Review by gkalpak — APPROVED** [review]
👍

File: aio/content/guide/testing.md [hunk 3]
```diff
@@ -3044,9 +2866,8 @@ Here are the most important static methods, in order of likely utility.
       The `TestBed.get` method takes an optional second parameter,
       the object to return if Angular can't find the provider
       (`null` in this example):
-      <code-example path="testing/src/app/bag/bag.spec.ts" region="testbed-get" title="src/app/bag/bag.spec.ts" linenums="false">
 
-      </code-example>
+      <code-example:></code-example> path="testing/src/app/bag/bag.spec.ts" region="testbed-get" title="src/app/bag/bag.spec.ts" linenums="false"></code-example>
```
[the remark this answers is not in the document]
Comment (petebacondarwin) [reply aio/content/guide/testing.md]: crap!

**Comment by petebacondarwin** [comment]
This commits of this PR was included in #20038, which has been merged

**Pull request closed without merging**